lib/trip_dispatcher.rb

```diff
@@ -60,17 +91,77 @@ def find_passenger(id)
       return @passengers.find { |passenger| passenger.id == id }
     end
 
+    def find_driver(id)
+      check_id(id)
+      return @drivers.find { |driver| driver.id == id }
+    end
+
     def inspect
       return "#<#{self.class.name}:0x#{self.object_id.to_s(16)} \
               #{trips.count} trips, \
               #{drivers.count} drivers, \
               #{passengers.count} passengers>"
     end
 
-    private
+    # private
 
     def check_id(id)
       raise ArgumentError, "ID cannot be blank or less than zero. (got #{id})" if id.nil? || id <= 0
     end
+
+
+
+    # Note: The user_id must correspond to a User instance
+   def request_trip(user_id)
+     # Finding the User instance
+     current_passenger = @passengers.find { |passenger| passenger.id == user_id }
+
+     if current_passenger == nil
+       raise ArgumentError.new("There is no user instance with the id #{user_id}")
+     end
+
+
+     # Checking for available drivers that do not have the same id as the user requesting a ride
+     available_drivers = []
+     @drivers.each do |x|
+       if x.status == :AVAILABLE && x.id != user_id.to_i
+         available_drivers << x
+       end
+     end
+
+     # If there are available drivers, then create trip instance.
+     if available_drivers.length > 0
+       # number_of_drivers = available_drivers.length
+       # random_driver = available_drivers[rand(0...number_of_drivers)]
+       # random_driver_id = random_driver.id
+       first_available_driver = available_drivers.first
+
+       trip_info = {
+         id: @trips.length + 1,
+         passenger: current_passenger,
+         start_time: Time.now,
+         end_time: nil,
+         cost: nil,
+         rating: nil,
+         driver: first_available_driver
+       }
+
+       trip = Trip.new(trip_info)
+       @trips << trip
+
+       # Changing driver data
+       first_available_driver.add_driven_trip(trip)
+       first_available_driver.change_availability
+
+       # Changing current passenger data
+       current_passenger.add_trip(trip)
+
+       return trip
+
+     else
+       return "No available drivers"
+       # return nil
```

Comment:
You should either return `nil` or raise an exception here, not return a string. There are a number of reasons for this, but ultimately what it comes down to is, in order to check that a string contains an error message, you need to read it.